### ISSUE TITLE
aws-lc-sys leaks source paths bypassing rustc's trim-paths

### ISSUE THREAD

**Issue by kinddstranger** [aws/aws-lc-rs#470]
## Repro

```toml
[profile.release]
strip = "symbols"
trim-paths = "all"

[dependencies]
aws-lc-rs = "1"
```

1. `$ cargo build --release`
2. `$ strings ./target/release/executable | grep $USER`
3. Encounter similar output:
```
/home/myusername/.cargo/registry/src/index.crates.io-6f17d22bba15001f/aws-lc-sys-0.20.0/aws-lc/crypto/bytestring/cbb.c
/home/myusername/.cargo/registry/src/index.crates.io-6f17d22bba15001f/aws-lc-sys-0.20.0/aws-lc/crypto/cipher_extra/e_aesgcmsiv.c
/home/myusername/.cargo/registry/src/index.crates.io-6f17d22bba15001f/aws-lc-sys-0.20.0/aws-lc/crypto/cipher_extra/e_chacha20poly1305.c
/home/myusername/.cargo/registry/src/index.crates.io-6f17d22bba15001f/aws-lc-sys-0.20.0/aws-lc/crypto/ecdsa_extra/ecdsa_asn1.c
/home/myusername/.cargo/registry/src/index.crates.io-6f17d22bba15001f/aws-lc-sys-0.20.0/aws-lc/crypto/ec_extra/ec_asn1.c
...
```

---

Why this behavior is undesired: Some crates may prefer that their release binaries do not contain any references to source paths on the builder's machine, at least because those will include his username.

I couldn't find out what part of your codegen puts these paths in binaries, but I'm hopeful that someone can.

**Comment by justsmth** [comment]
I put up a draft PR for this issue. I tested the build locally on my Mac and it works as expected:
```
aws-lc-rs on  strip-release-binary [$] via 🦀 v1.80.0
at 15:21:44 ❯ strings `find target/release -name "lib*_crypto.*"` | grep 'bn/bn.c'
/aws-lc/crypto/fipsmodule/bn/bn.c

aws-lc-rs on  strip-release-binary [$] via 🦀 v1.80.0
at 15:26:42 ❯ strings `find target/release -name "lib*_crypto.*"` | grep 'justsmth'
```

Let us know any comments/concerns you have about the approach used. Feel free to pull the PR branch locally for testing.

**Comment by kinddstranger** [comment]
Good work! Looking forward to having it merged

**Comment by justsmth** [comment]
This is resolved with the release of [aws-lc-rs v1.9.0](https://github.com/aws/aws-lc-rs/releases/tag/v1.9.0).
